internal/config/config.go

```diff
@@ -468,36 +470,43 @@ func peerFromCR(p metallbv1beta2.BGPPeer, passwordSecrets map[string]corev1.Secr
 		RouterID:      routerID,
 		NodeSelectors: nodeSels,
 		Password:      password,
+		PasswordRef:   passwordRef,
 		BFDProfile:    p.Spec.BFDProfile,
 		EBGPMultiHop:  p.Spec.EBGPMultiHop,
 		VRF:           p.Spec.VRFName,
 	}, nil
 }
 
-func passwordForPeer(p metallbv1beta2.BGPPeer, passwordSecrets map[string]corev1.Secret) (string, error) {
+func passwordForPeer(p metallbv1beta2.BGPPeer, passwordSecrets map[string]corev1.Secret) (string, corev1.SecretReference, error) {
```

Comment:
At the cost of doing the same thing twice, I'd split this in "passwordForPeer" and "passwordSecretForPeer".
The two return values is not super clear

Reply:
went for something a bit different, defaulting to the peer's password and unpacking the secret when specified